### PR TITLE
Ee 4471/new api

### PR DESCRIPTION
**Pull request by mikehalmaho**
Convert the existing app to work on the new API.  No new functionality has been added other than adopting the new API and making sure that the existing tests work.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/proving/income/api/Applicant.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package uk.gov.digital.ho.proving.income.api;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.LocalDate;
+
+public class Applicant {
```
Comment (jhevans) [inline src/main/java/uk/gov/digital/ho/proving/income/api/Applicant.java]: Could use Lombok here for the getters and constructor, same goes for other POJOs. It is specified in the [coding standards](https://confluence.ipttools.info/pages/viewpage.action?pageId=114222496) but it's open to interpretation whether it's mandatory or recommended.

File: src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusCheckResponseV2.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package uk.gov.digital.ho.proving.income.api;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import uk.gov.digital.ho.proving.income.domain.Individual;
+
+@Deprecated
```
Comment (jhevans) [inline src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusCheckResponseV2.java]: I assume the deprecated classes are/could be still in use by someone, i.e they can't just be removed?

File: src/test/java/uk/gov/digital/ho/proving/income/JacksonTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.time.Month;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+/**
+ * I've decided to keep this test as it is useful for debugging Jackson marshaling / unmarshaling errors in other tests
+ */
```
Comment (jhevans) [inline src/test/java/uk/gov/digital/ho/proving/income/JacksonTest.java]: It's an odd test to have, was this something you created for debugging purposes in the first place?

**Review by jhevans — COMMENTED** [review]
Looks good on the whole, just a couple of comments and possible nice to haves.  It's a hefty one though so worth getting a second opinion.

File: src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusCheckResponseV2.java (hunk 2 above)
Comment (mikehalmaho) [reply src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusCheckResponseV2.java]: I believe its only in use by the UI, but by keeping the old endpoint for now it means we do not have to synchronise the deployment with the UI changes.  I fully expect to remove this endpoint before the dev is complete.

File: src/main/java/uk/gov/digital/ho/proving/income/api/Applicant.java (hunk 1 above)
Comment (mikehalmaho) [reply src/main/java/uk/gov/digital/ho/proving/income/api/Applicant.java]: Good call.  This project has now been Lomboked to within an inch of its life.

File: src/test/java/uk/gov/digital/ho/proving/income/JacksonTest.java (hunk 3 above)
Comment (mikehalmaho) [reply src/test/java/uk/gov/digital/ho/proving/income/JacksonTest.java]: I ended up writing the same class twice (for debugging purposes) so I wanted a copy of it saving somewhere.  But yes it is odd, so I have deleted it and can always pull it out of the history if required.

File: src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusService.java [hunk 4]
```diff
@@ -46,34 +46,50 @@ public FinancialStatusService(HmrcClient hmrcClient, AuditClient auditClient, Ni
...
+        FinancialStatusCheckResponseV2 responseV2 = new FinancialStatusCheckResponseV2(response.status(), response.individuals().get(0), categoryCheckV2);
+        return responseV2;
+    }
+
+    @PostMapping(value = "/incomeproving/v3/individual/financialstatus", produces = APPLICATION_JSON_VALUE)
     public FinancialStatusCheckResponse getFinancialStatus(@Valid @RequestBody FinancialStatusRequest request) {
 
-        final String redactedNino = ninoUtils.redact(request.getNino());
-        log.info("Financial status check request received for {} - applicationRaisedDate = {}, dependents = {}", redactedNino, request.getApplicationRaisedDate(), request.getDependants());
+        Applicant mainApplicant = request.applicants().get(0);
```
Comment (Tim-Hammonds) [inline src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusService.java:67]: The request is checked with @Valid - should the class have a not null validation on applicants to make this statement null safe? Is it possible to be called with no applicants?

Comment (mikehalmaho) [reply src/main/java/uk/gov/digital/ho/proving/income/api/FinancialStatusService.java:67]: Agreed - I will take this up on the next piece of dev.

File: src/main/java/uk/gov/digital/ho/proving/income/domain/Individual.java [hunk 5]
```diff
@@ -1,44 +1,28 @@
...
+import lombok.ToString;
+import lombok.experimental.Accessors;
+
+@AllArgsConstructor
+@Getter
+@Accessors(fluent = true)
+@ToString
 public class Individual {
-    private String title;
+    @JsonProperty
```
Comment (Tim-Hammonds) [inline src/main/java/uk/gov/digital/ho/proving/income/domain/Individual.java:14]: I thought we were going to add the name of the json field to this annotation. If we don't the annotation is (sort of) superfluous.

**Review by Tim-Hammonds — APPROVED** [review]

File: src/main/java/uk/gov/digital/ho/proving/income/domain/Individual.java (hunk 5 above)
Comment (mikehalmaho) [reply src/main/java/uk/gov/digital/ho/proving/income/domain/Individual.java:14]: Done, plus added serialization tests.

**Pull request merged**